### PR TITLE
docs(examples): add Gatsby example

### PR DESCRIPTION
**Pull request by bweston92**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

Add an example for Gatsby sites.

## Other information

Currently struggling to get it working.

### CODE REVIEW

File: examples/gatsby/.bazelrc [hunk 1]
```diff
@@ -0,0 +1 @@
+common --experimental_allow_incremental_repository_updates
```
Comment (alexeagle) [inline examples/gatsby/.bazelrc]: see other examples - this ought to include our common bazelrc

File: examples/gatsby/.gitignore [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Logs
+logs
+*.log
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+
+# Runtime data
```
Comment (alexeagle) [inline examples/gatsby/.gitignore]: reduce this file down to the lines needed in the example

File: examples/gatsby/.prettierrc [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "endOfLine": "lf",
```
Comment (alexeagle) [inline examples/gatsby/.prettierrc]: I don't think it's illustrative to have a formatter in this example, it's orthogonal to what we want to show

File: examples/gatsby/BUILD.bazel [hunk 4]
```diff
@@ -0,0 +1,27 @@
+load("@build_bazel_rules_nodejs//:defs.bzl", "nodejs_binary")
+
+nodejs_binary(
```
Comment (alexeagle) [inline examples/gatsby/BUILD.bazel]: should be

```
load("@npm//gatsby-cli:index.bzl", "gatsby")
```
which is a nodejs_binary already set up to run that bin

**Comment by bweston92** [comment]
It shouldn't be green, it doesn't run. I will update tomorrow.

File: examples/gatsby/.prettierrc (hunk 3 above)
Comment (bweston92) [reply examples/gatsby/.prettierrc]: Whoops that is from the `gatsby new` tool.

Comment (bweston92) [reply examples/gatsby/.prettierrc]: deleted.

**Comment by alexeagle** [comment]
to fix the CI you need to `yarn bazel:update-deleted-packages`
(see comment in /.bazelrc)

File: examples/gatsby/BUILD.bazel [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    name = "build",
+    args = ["build"],
+    data = [
+        ":src",
+        "@npm//:node_modules",
+    ],
+)
+
+filegroup(
+    name = "src",
```
Comment (alexeagle) [inline examples/gatsby/BUILD.bazel]: this rule comes first in the depgraph so I'd usually have it come first in the file

File: examples/gatsby/BUILD.bazel [hunk 6]
```diff
@@ -0,0 +1,25 @@
+load("@npm//gatsby-cli:index.bzl", "gatsby")
+
+gatsby(
```
Comment (alexeagle) [inline examples/gatsby/BUILD.bazel]: this has no outs[] or output_dir which means it doesn't do anything unless you `bazel run :build` - is that what you intend? how would you access the built artifacts? I would expect this target to be a dependency of a devserver or something

File: examples/gatsby/BUILD.bazel [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+            "config/*.js",
+            "plugins/**/*",
+            "src/**/*",
+            "gatsby-*.js",
+        ],
+        exclude = ["node_modules/**"],
+    ) + [
+        "package.json",
+    ],
+)
```
Comment (alexeagle) [inline examples/gatsby/BUILD.bazel]: can you add some test that proves we did the right thing?

File: examples/gatsby/BUILD.bazel (hunk 6 above)
Comment (bweston92) [reply examples/gatsby/BUILD.bazel]: Once I got the build working I would progress to that yes, at the minute can't get the build working.

**Pull request closed without merging**